Content.IntegrationTests/Tests/Round/CharacterSelectionTest.cs

```diff
@@ -0,0 +1,481 @@
+                new() { Jobs = [ Captain ] },
+                new() { Jobs = [ Clown ] }
+            ],
+            ExpectedJob = Passenger
+        },
+        // adapted from https://github.com/space-wizards/space-station-14/pull/36493#issuecomment-2926983119
+        // Since the game mode tries to make every player an antag, Captain should be excluded even if traitor is
+        // selected on the Captain character.
+        new()
+        {
+            Description = "Antag chars, one command",
+            MediumPrioJobs = [ Captain, Passenger ],
+            Characters =
+            [
+                new() { Jobs = [ Captain ], IsTraitor = true },
+                new() { Jobs = [ Passenger ], IsTraitor = true, ExpectToSpawn = true }
+            ],
+            ExpectedJob = Passenger,
+            ExpectTraitor = true
+        },
+        // also adapted from https://github.com/space-wizards/space-station-14/pull/36493#issuecomment-2926983119
+        // BUT the behaviour described there was later changed as described in case 1 in
+        // https://github.com/space-wizards/space-station-14/pull/36493#issuecomment-3014257219,
+        // so this tests the updated behaviour
+        // Since this player has no enabled characters that are eligible to be a traitor, their session will not be
+        // selector to be an antag, and thus the Captain character will be eligible to spawn, even if the Captain
+        // character has traitor enabled.
+        new()
+        {
+            Description = "Only antag captain enabled",
+            MediumPrioJobs = [ Captain, Passenger ],
+            Characters =
+            [
+                new() { Jobs = [ Captain ], IsTraitor = true, ExpectToSpawn = true },
+                new() { Jobs = [ Passenger ], IsTraitor = true, Enabled = false }
+            ],
+            ExpectedJob = Captain
+        },
+        // Case 1 from https://github.com/space-wizards/space-station-14/pull/36493#issuecomment-3014257219
+        // if a player has no antag-compatible jobs enabled they should not be selected as an antag
+        new() {
+            Description = "Only antag captain",
+            HighPrioJob = Captain,
+            Characters =
+            [
+                new() { Jobs = [ Captain ], IsTraitor = true, ExpectToSpawn = true }
+            ],
+            ExpectedJob = Captain,
+            ExpectTraitor = false
+        },
+        // Case 2 from https://github.com/space-wizards/space-station-14/pull/36493#issuecomment-3014257219
+        // a player that is selected as an antag should always roll a character & job compatible with that
+        // antag selection
+        // Put another way, a player should only be pre-selected to be an antag if they have characters to be eligible
+        // to be an antag. Thus if a player is pre-selected to be an antag, they should then only be eligible for the
+        // set of antag-compatible jobs made from enabled characters.
```

Comment:
Suggest changing "enabled characters" to "enabled characters that have that antag enabled."